graylog2-server/src/main/java/org/graylog2/lookup/adapters/CSVFileDataAdapter.java

```diff
@@ -82,6 +86,9 @@ public void doStart() throws Exception {
         if (isNullOrEmpty(config.path())) {
             throw new IllegalStateException("File path needs to be set");
         }
+        if (!pathChecker.fileIsInAllowedPath(Paths.get(config.path()))) {
+            throw new IllegalStateException("The specified CSV file is not in an allowed path.");
+        }
```

Comment:
I just noticed that we are missing this check in `doRefresh()`
So, if you wait after the check interval of the data adapter, the file gets loaded anyway.

Reply:
Good find. Addressed in https://github.com/Graylog2/graylog2-server/pull/10244/commits/edb88c44e1a1d7fa067baf9acbb21ab72142bf20